src/pages/resultsView/fusion/ResultViewFusionMapperStore.ts

```diff
@@ -0,0 +1,212 @@
+     * @type {MobxPromiseUnionTypeWithDefault<Array<EnsemblTranscript>>}
+     */
+    readonly ensemblTranscripts = remoteData({
+        await: () => [
+            this.fusionData,
+        ],
+        invoke: async () => {
+            // Collect all transcript id's from the fusion data
+            let transcriptIds: any = [];
+            if (this.fusionData.result) {
+                this.fusionData.result.forEach((fusion: StructuralVariant) => {
+                    if (transcriptIds.indexOf(fusion.site1EnsemblTranscriptId) < 0) {
+                        transcriptIds.push(fusion.site1EnsemblTranscriptId);
+                    }
+                    if (transcriptIds.indexOf(fusion.site2EnsemblTranscriptId) < 0) {
+                        transcriptIds.push(fusion.site2EnsemblTranscriptId);
+                    }
+                });
+            }
+            // Fetch ensembl transcript data with transcript ids as request params
+            return await fetchEnsemblTranscriptsByEnsemblFilter({'transcriptIds': transcriptIds})
+                .then((result: EnsemblTranscriptExt[]) => {
+                    let fusionTranscripts: EnsemblTranscriptExt[] = [];
+                    let refTranscripts: EnsemblTranscriptExt[] = [];
+                    return result.map(t => {
+                        // Check if transcript is reference gene
+                        t.isReferenceGene = t.hugoSymbols.indexOf(this.gene.hugoGeneSymbol) >= 0;
+                        this.applyColor(t, t.isReferenceGene, t.isReferenceGene ? refTranscripts : fusionTranscripts);
+                        return t;
+                    });
+                });
+        }
+    }, []);
+
+    /**
+     * Fetch pfam domain data
+     * @type {MobxPromiseUnionTypeWithDefault<PfamDomain[]>}
+     */
+    readonly pfamDomainData = remoteData<PfamDomain[] | undefined>({
+        await: () => [
+            this.ensemblTranscripts
+        ],
+        invoke: async () => {
+            if (this.ensemblTranscripts.result) {
+                // get all pfam domain ids from requested transcripts
+                let pFamDomainIds = this.ensemblTranscripts.result
+                    .map(t => t.pfamDomains)
+                    .reduce((accumulator, currentValue) => accumulator.concat(currentValue), [])
+                    .map(t => t.pfamDomainId);
+                // fetch the data with pfam domain ids as request params
+                return await fetchPfamDomainData(pFamDomainIds);
+            }
+        }
+    }, undefined);
+
+    /**
+     * Get number of occurences
+     * @returns {{[p: string]: number}}
+     */
+    @computed
+    get fusionOccurrence(): { [T: string]: number } {
+        return _.countBy(this.fusionData.result, (d) => d.label);
+    }
+
+    /**
+     * Get pfam domain data
+     * @returns {PfamDomain[]}
+     */
+    @computed
+    get processedPfamDomains(): PfamDomain[] {
+        return this.pfamDomainData.result || [];
```

Comment:
whats the purpose of this? users should await pfamDomainData, or pfamDomainData should have default value set of []

Reply:
same goes for `processedEnsembleTranscripts`